src/main/java/com/redhat/devtools/intellij/knative/actions/RefreshAction.java

```diff
@@ -36,21 +43,32 @@ public RefreshAction(Class... filters) {
 
     @Override
     public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected) {
+        TelemetryMessageBuilder.ActionMessage telemetry = TelemetryService.instance().action(NAME_PREFIX_MISC + "refresh");
         Tree tree = getTree(anActionEvent);
         if (tree == null) {
+            ExecHelper.submit(() -> telemetry
```

Comment:
I don't think you need this because the telemetry already uses a thread pool to handle requests. Better asking confirmation by @adietish

Reply:
By looking at tekton i guess you're right. Removed it.